save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/UsersController.kt

```diff
@@ -68,8 +61,8 @@ class UsersController(
     @GetMapping("/find-by-name/{userName}")
     fun findByName(
         @PathVariable userName: String,
-    ): Mono<StringResponse> = userService.findByName(userName).map {
-        ResponseEntity.ok().body(springUserDetailsWriter.writeValueAsString(it))
+    ): Mono<AuthenticationUserDetailsResponse> = userService.findByName(userName).map {
+        ResponseEntity.ok().body(AuthenticationUserDetails(it))
```

Comment:
Ah, you moved it from a simple string to some data class, that's why you need new UserDetails for authorization? (this is how I get it)

Reply:
yes, but here is another change:
I don't why but we didn't use the Spring engine to parse Object provided by backend to api-gateway. We created a custom `objectMapper` to decode it from String.
I guess the reason is in `getters`. Even in current implementation I have to add `@JsonIgnore` in `AuthenticationUserDetails`. Previously we used `org.springframework.security.core.userdetails.User` and additionally registered `org.springframework.security.jackson2.SecurityJackson2Modules` (before it we have a custom extension for `SpringUser` and a custom serializer for it)